Add a mastodon link

laurentperrinet opened the pull request:
Replaced the display to my twitter account to the mastodon one.

- imported the (openly available) Mastodon logo from https://commons.wikimedia.org/wiki/File:Mastodon_Logotype_(Simple).svg
- added the jinja template to `content/00.front-matter.md`
- testing with a dummy account @JohnMastodon@mastodon.social

agitter commented:
Supporting Mastodon is a good idea. Enough scientists have migrated that there will be reasonable demand.

Two comments before reviewing this in detail:
- Please update add these optional metadata fields to `USAGE.md` (https://github.com/manubot/rootstock/blob/main/USAGE.md#manuscript-metadata)
- What is the example account being used? It looks like a real account. We may want to use a johndoe account instead of a real Mastodon account but should discuss this.

laurentperrinet commented:
I have 
* updated the `USAGE.md` file
* replaced `JohnMastodon` (which is a known mock account created by Eugen Rochko to circumvent the ban of the "Join Mastodon" phrase) by `johndoe`

dhimmel commented on USAGE.md:
```diff
@@ -236,6 +236,8 @@ name: Daniel S. Himmelstein  # mandatory
 orcid: 0000-0002-3012-7446  # mandatory
 twitter: dhimmel  # optional
+mastodon: dhimmel  # optional: mastodon username
+mastodon: mastodon.social  # optional: mastodon server (instance)
```
Typo here: `mastodonserver`

dhimmel replied:
I think we probably want to use `mastodon-server` as the key versus `mastodonserver` since that's the style [used elsewhere](https://github.com/manubot/manubot/blob/82f25cb8d698a12d69e533686867a8a4afd35c4c/manubot/pandoc/tests/test_cite_filter/input.md#L2-L22) by manubot and pandoc.

dhimmel replied:
Noting an experiment I did at https://genomic.social/@dhimmel/109710993689898079

dhimmel replied:
@laurentperrinet would you be able to update the `mastodonserver` key to `mastodon-server` and fix line 240? I think that is the last outstanding required change on this PR.

dhimmel commented on content/images/mastodon.svg:
```diff
@@ -0,0 +1,4 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="61.076954mm" height="65.47831mm" viewBox="0 0 216.4144 232.00976">
+  <path fill="#2b90d9" d="M211.80734 139.0875c-3.18125 16.36625-28.4925 34.2775-57.5625 37.74875-15.15875 1.80875-30.08375 3.47125-45.99875 2.74125-26.0275-1.1925-46.565-6.2125-46.565-6.2125 0 2.53375.15625 4.94625.46875 7.2025 3.38375 25.68625 25.47 27.225 46.39125 27.9425 21.11625.7225 39.91875-5.20625 39.91875-5.20625l.8675 19.09s-14.77 7.93125-41.08125 9.39c-14.50875.7975-32.52375-.365-53.506 ...
```
We should make this greyscale using the same grey color as the other logos.

dhimmel commented on content/metadata.yaml:
```diff
@@ -11,6 +11,8 @@ authors:
     initials: JD
     orcid: XXXX-XXXX-XXXX-XXXX
     twitter: johndoe
+    mastodon: johndoe
+    mastodonserver: mastodon.social
```
I think it might be preferable to have a single field like `mastodon: johndoe@mastodon.social`. https://bioregistry.io/ doesn't have a `mastodon` record yet, so not sure where to link regarding advice on the proper CURIE format.

Is there a way to resolve a mastodon username@server to direct to the host server?

Aside: IMO the federated model of mastodon leads to a pretty bad user experience. And seems like there will be a lot of tootrot as servers are taken offline. Not against adding links though since there does seem to be adoption among scholars.

dhimmel commented:
How the logo renders looks good:

> ![image](https://user-images.githubusercontent.com/1117703/213933241-6fd9c788-421b-4296-8230-9bfa678ee76a.png)

dhimmel commented:
Ah noting this comment by @agitter:

> Please update add these optional metadata fields to USAGE.md (https://github.com/manubot/rootstock/blob/main/USAGE.md#manuscript-metadata)

agitter commented on USAGE.md:
```diff
@@ -236,6 +236,8 @@ name: Daniel S. Himmelstein  # mandatory
 orcid: 0000-0002-3012-7446  # mandatory
 twitter: dhimmel  # optional
+mastodon: dhimmel  # optional: mastodon username
+mastodon-server: mastodon.social  # optional: mastodon server (instance)
```
```suggestion
mastodon-server: genomic.social  # optional: mastodon server (instance)
```
I believe this was the previous suggestion to use @dhimmel's real account in the example metadata

agitter reviewed (commented):
This is my only other small suggestion. Once we have the build working it looks good to me.

Thanks for the contribution @laurentperrinet!

agitter commented on content/00.front-matter.md:
```diff
@@ -39,6 +39,10 @@ on {{manubot.date}}.
+  {%- endif %}
+  {%- if author.mastodon is defined and author.mastodon is not none and author["mastodon-server"] is defined and author["mastodon-server"] is not none %}
+  · ![Mastodon icon](images/mastodon.svg){.inline_icon}
+  [\@{{author.mastodon}}@{{author.mastodon-server}}](https://{{author["mastodon-server"]}}/@{{author.mastodon}})
```
```suggestion
  [\@{{author.mastodon}}@{{author["mastodon-server"]}}](https://{{author["mastodon-server"]}}/@{{author.mastodon}})
```
I think the build failed again because we missed this one

laurentperrinet commented:
It seems that this PR went to the `master` branch instead of `main`. Do you want me to make a new PR?

dhimmel commented:
Oops! No let's see if I can cherry pick the commit and put it on `main`.

dhimmel commented:
There were conflicts so I opened a PR just so we can review and make sure CI looks good https://github.com/manubot/rootstock/pull/486
